### ISSUE TITLE
[BUG] After installing drivers, bluetooth no longer works

### ISSUE THREAD

**Issue by DyingDucks** [Davidobot/BetterJoy#943]
**Describe the bug**
I installed ViGEM and HIDGuardian. After installation I opened BetterJoy and added my controller. After adding my controller my mouse was going crazy and I restarted my computer. After rebooting the pc, my Bluetooth no longer works at all on my computer, I can't add or remove any devices as it says my Bluetooth is nonexistent. I'm on Windows 11. My device manager also keeps refreshing as if a device is being added and unadded repeatedly 

**Controller (please complete the following information):**
 - Type: Pro GameCube Controller
 - Connection: BT

**Comment by felipecardona91** [comment]
hello, did you fix it_? i am having the same issue

**Comment by plxjammerplx** [comment]
I had a similar issue in the past, the only way I fixed it was to manually remove the wifi/bt card and reseat it back into the slot. I haven't had this issue since version 7.

**Comment by felipecardona91** [comment]
> 

how can i manually remove a wifi/bt card if it is attached to the motherboard?

**Comment by plxjammerplx** [comment]
> > 
> 
> how can i manually remove a wifi/bt card if it is attached to the motherboard?

It should be a [small m.2 sized card](https://www.google.com/search?q=intel+wifi+card&sxsrf=APq-WBtx6N5wngwzunooAKz1GV_9TdfgQA:1646525606904&source=lnms&tbm=isch&sa=X&ved=2ahUKEwj9xKbvmbD2AhUtTd8KHaCGBowQ_AUoAnoECAIQBA&biw=2560&bih=1289&dpr=1#imgrc=9JImCRkgb06l9M) attached to your motherboard held down by a screw.

**Comment by felipecardona91** [comment]
> > > 
> > 
> > 
> > how can i manually remove a wifi/bt card if it is attached to the motherboard?
> 
> It should be a [small m.2 sized card](https://www.google.com/search?q=intel+wifi+card&sxsrf=APq-WBtx6N5wngwzunooAKz1GV_9TdfgQA:1646525606904&source=lnms&tbm=isch&sa=X&ved=2ahUKEwj9xKbvmbD2AhUtTd8KHaCGBowQ_AUoAnoECAIQBA&biw=2560&bih=1289&dpr=1#imgrc=9JImCRkgb06l9M) attached to your motherboard held down by a screw.
> 
> Some motherboards will have a [cover for their wifi/bt cards](https://static.tweaktown.com/content/8/7/8718_21_asrock-fatal1ty-b450-gaming-itx-ac-motherboard-review_full.jpg) you need to unscrew that cover and lift open the cover to have access to your wifi/bt card or it's just unscrewing the cover from the motherboard and pulling it out directly. Just double check to see which one you have.
> 
> Note: If it's the one with the silver cover you will be required to unscrew the antennas from the i/o shield prior to taking out the wifi card.
> 
> If you still need further help, post images of where your wifi/bt card is located at on the motherboard.

i don't know what i did, but i removed gpu, disconected every usb, disconected usb hub, removed antennas, the wifi antenna was really deep in my motherboad, removing it would have been a pain in the ....., so i just... wiggled it a little bit, moved it, but, pretty much sure i did nothing, it did not have enough room to move, long story short, ended up making a maintance to my dusty pc, put it back together...and boom, fixed, magically...weirdest thing ever. thank you very much because i am sure you had something to do with it. 

btw, my board is a asrock z490m-itx/ac wifi

**Comment by plxjammerplx** [comment]
> > > > 
> > > 
> > > 
> > > how can i manually remove a wifi/bt card if it is attached to the motherboard?
> > 
> > 
> > It should be a [small m.2 sized card](https://www.google.com/search?q=intel+wifi+card&sxsrf=APq-WBtx6N5wngwzunooAKz1GV_9TdfgQA:1646525606904&source=lnms&tbm=isch&sa=X&ved=2ahUKEwj9xKbvmbD2AhUtTd8KHaCGBowQ_AUoAnoECAIQBA&biw=2560&bih=1289&dpr=1#imgrc=9JImCRkgb06l9M) attached to your motherboard held down by a screw.
> > Some motherboards will have a [cover for their wifi/bt cards](https://static.tweaktown.com/content/8/7/8718_21_asrock-fatal1ty-b450-gaming-itx-ac-motherboard-review_full.jpg) you need to unscrew that cover and lift open the cover to have access to your wifi/bt card or it's just unscrewing the cover from the motherboard and pulling it out directly. Just double check to see which one you have.
> > Note: If it's the one with the silver cover you will be required to unscrew the antennas from the i/o shield prior to taking out the wifi card.
> > If you still need further help, post images of where your wifi/bt card is located at on the motherboard.
> 
> i don't know what i did, but i removed gpu, disconected every usb, disconected usb hub, removed antennas, the wifi antenna was really deep in my motherboad, removing it would have been a pain in the ....., so i just... wiggled it a little bit, moved it, but, pretty much sure i did nothing, it did not have enough room to move, long story short, ended up making a maintance to my dusty pc, put it back together...and boom, fixed, magically...weirdest thing ever. thank you very much because i am sure you had something to do with it.
> 
> btw, my board is a asrock z490m-itx/ac wifi

Yup, I have the Asrock B450 Fatality itx board with the same silver cover wifi/bt card. This happened to me several times in the past and reseating the wifi/bt card was the only way for me to fix the weird random missing BT issue. Annoying way to fix this issue but it works.

**Comment by weishen1121** [comment]
I have a X570 AORUS ELITE WIFI motherboard and I've looked through the manual but it does not seem to have a removable option for bluetooth or wireless module. It seems to be located at the top left according to the manual. I also looked through the BIOS but no option to disable wireless or bluetooth. Any other ideas?

My issue is that it keeps toggling on/off and needs to be disabled in Device Manager to stop. 

I think BetterJoy needs to be shut down if it can cause such an issue... this has been melting my brain for nearly 3 hours now UGH

**Comment by felipecardona91** [comment]
what solved it for me, after days, i don't understand why, was just
disconnecting the wifi/BT antennas, opening my computer and kinda...moving
the antennas receptor a bit, sounds stupid, i know, but it worked.

**Comment by weishen1121** [comment]
> what solved it for me, after days, i don't understand why, was just disconnecting the wifi/BT antennas, opening my computer and kinda...moving the antennas receptor a bit, sounds stupid, i know, but it worked. 
> Well, your motherboard has an io cover on top of the wifi/bt card. To get access to your wifi/bt card on that particular motherboard, you would need to remove the screws on the back of the motherboard that's holding the io cover in place. There should be like 3-6 screws on the back of the motherboard(varies) holding the io shield and the io cover together. Then another 2 screws holding the wifi module in the m.2 key slot.
> 
> I have labelled some of the screws on the back side of the motherboard: https://imgur.com/a/zIg1XkN
> 
> Since I don't have your board in hand, I'm not sure if the io cover and vrm heatsink cover is connected together. Also when you do remove the covers, do take you time since I think there might also be an LED cable underneath the io shield cover connected to the motherboard.
Thanks for the responses!
Also thank you for the picture with detailed instructions for my I/O shield removal, really appreciate it.

I had read yesterday that shutting down the computer and unplugging the power worked for some, so I tried it yesterday but only shutting it down for a few minutes at most. I had read a post where someone said 24 hours but I was very doubtful. I still went ahead and shut it down last night and unplugged it while I slept. It's been about 8 hours. When I turned it on this morning, my Intel Wireless Bluetooth entry under Device Manager was still disabled, so I uninstalled + deleted software in one last attempt. Scanned for hardware changes and bam, it actually worked without disconnecting/connecting again! I had also removed all my paired bluetooth devices and they were gone, but as soon as I scanned for hardware changes, they all came back and all listed under the same bluetooth section. (the bluetooth section only had the Intel Wireless Bluetooth entry under Device Manager previously) Just to be sure, I removed all my bluetooth pairings and restarted the computer one last time. Paired my DS from PS5 and everything seems to be working! Was even able to turn on Hide DS4 Controller option under DS4windows with no issues! 

I'm not exactly sure how this works out, but hopefully this will help someone else in this position as well. 
Just leave it alone, unplugged from the power for ~8 hours and hopefully won't need to go insane troubleshooting for 4+ hours lmao.
